DependencyInjectionContainer/DIContainer.cs

```diff
@@ -1,130 +1,228 @@
+                if (!IsParentContainerExist())
+                {
+                    throw new NullReferenceException("Current container do not have parent");
+                }
+                return сontainerParent.Resolve(typeToResolve, ResolveStrategy.Any);
+
+            case ResolveStrategy.Any:
+                if (TryGetRegistration(typeToResolve, ResolveStrategy.Any, out Service foundService))
+                {
+                    return GetOrCreateServiceImplementation_SaveIfSingleton(foundService, resolveSource);
+                }
+                throw new ServiceNotFoundException(typeToResolve);
+
+            case ResolveStrategy.Local:
+                if (TryGetRegistration(typeToResolve, ResolveStrategy.Local, out foundService))
+                {
+                    return GetOrCreateServiceImplementation_SaveIfSingleton(foundService, resolveSource);
+                }
+                throw new ServiceNotFoundException(typeToResolve);
+
+            default:
+                throw new ArgumentException("Wrong resolve source type");
+        }
+    }
 
-            if(serviceToResolve.Implementation != null)
-            {
-                return serviceToResolve.Implementation;
-            }
+    private bool TryGetRegistration(Type typeForSearch, ResolveStrategy resolveSource, out Service foundService)
+    {
+        List<Service> servicesImplementsType = new List<Service>();
 
-            var implementation = GetCreatedImplementationForService(serviceToResolve);
+        switch (resolveSource)
+        {
+            case ResolveStrategy.Local:
+                servicesImplementsType = registeredServices
+                                         .Where(service => service.Key == typeForSearch)
+                                         .ToList();
+                break;
+            case ResolveStrategy.Any:
+                servicesImplementsType = GetRegistrationIncludingParentContainer();
+                break;
+            case ResolveStrategy.NonLocal:
+                break;
+            default:
+                throw new ArgumentException("Wrong resolve source type");
+        }
 
-            if(serviceToResolve.Lifetime == ServiceLifetime.Singleton)
-            {
-                serviceToResolve.Implementation = implementation;
-            }
+        if (servicesImplementsType.Count > 1)
+        {
+            throw new ResolveServiceException($"Many services with type {typeForSearch} was registered. Use ResolveMany to resolve them all");
+        }
 
-            return implementation;
+        foundService = servicesImplementsType.SingleOrDefault();
+        return IsServiceFound();
 
-            object GetCreatedImplementationForService(Service service)
+        List<Service> GetRegistrationIncludingParentContainer()
+        {
+            var curContainer = this;
+            List<Service> found = new List<Service>();
+            while (curContainer != null && found.Count == 0)
```

Comment:
why found.Count == 0 ? It'll stop looking in parents if found something in current. Is that expected?